src/controllers/mainController.ts

```diff
@@ -1858,72 +1858,128 @@ export default class MainController implements vscode.Disposable {
         await vscode.env.openExternal(vscode.Uri.parse(Constants.gettingStartedGuideLink));
     }
 
+    private async newQueryFromProfile(
```

Comment:
`newQueryFromProfiles()` and `newQueryFromPrompt()` are mostly just extractions of the existing logic to helper functions for readability.

The only change should be where the `createObjectExplorerSession` flag is used.